components/engine_traits/src/write_batch.rs

```diff
@@ -123,4 +123,10 @@ pub trait WriteBatch: Mutable {
 
     /// Merge another WriteBatch to itself
     fn merge(&mut self, src: Self) -> Result<()>;
+
+    /// Set a custom sequence number for all of the writes in this batch. Used
+    /// by [`HybridEngine`]
+    fn set_sequence_number(&mut self, _seq: u64) -> Result<()> {
```

Comment:
Does it need to be in the trait? Or it can be the normal function in RegionCacheMemoryEngine::WriteBatch.

Reply:
I am leaning that way. Let me see if there are any downsides to this, and if there are not I will remove this.

Reply:
Done: this needed a minor re-factor to do cleanly, but I removed this from the WriteBatch API. Changed to `HybridEngineWriteBatchExt` be a partly specialized impl `HybridEngine<EK, RegionCacheMemoryEngine>`